### PR TITLE
Adding disabled property.

### PR DESCRIPTION
**Pull request by bernalrs**
Implementation of issue #4

### CODE REVIEW

**Comment by leMaik** [comment]
Is there any reason why you added eslint and a whole bunch of dependencies to implement the `disabled` property? We usually use [Standard Style](https://standardjs.com/) for our code.

File: src/components/SearchBar/SearchBar.js [hunk 1]
```diff
@@ -5,33 +5,103 @@ import SearchIcon from 'material-ui/svg-icons/action/search'
...
-  },
-  searchContainer: {
-    margin: 'auto 16px',
-    width: '100%'
+const getStyles = (props, state, context) => {
+  const { disabled }  = props;
+  const { value } = state;
+  const nonEmpty = value ? 0 : value.length
+
+  console.log(context.muiTheme.textField);
```
Comment (leMaik) [inline src/components/SearchBar/SearchBar.js]: This debug output should be removed.

Comment (bernalrs) [inline src/components/SearchBar/SearchBar.js]: Good catch

**Comment by bernalrs** [comment]
To answer your question previous question:

1. Using prettier for styling feel free to change the settings.
2. I understand that ESLint has more plugins and is the defacto standard these days but that might change;-).

**Comment by leMaik** [comment]
Allright, but we use neither of the tools, so I'd prefer to not having them in this project for now. :wink:  Especially formatting the code is something a good IDE should do, not an npm script.

**Comment by bernalrs** [comment]
In my opinion, IDE should not format code, people tend to use different IDE and therefore have different formatting rules. This does not take into account custom rules which will be hidden from contributors in some setting panel, instead, this should be made explicit within .editconfig and prettier.  
 
I understand not wanting to add tools you do not support might be an issue however not having code formatting added to the project as default adds a barrier of entry to code contributions outside of your company.  

Coincidentally,  these additions were suggested to me in a similar manner a few months ago and I was not entirely convinced at the time however this has greatly improved productivity.  Anyways let me know what you think I would be glad to go into more details on any of the tools.

**Comment by leMaik** [comment]
I _am_ convinced that tools are really useful to increase productivity. But this PR is about adding a `disabled` prop to a small component. I really like your first commit, which does precisely that. :+1: Re-formatting the entire code, adding loads of productivity boosting tools is just an overkill for a project of this size.

**Comment by saschb2b** [comment]
We really appreciate any quality increasing tipps and community growth strategies. But this PR was about adding a disabling functionality. It would be great if this PR just include adding/changing what's advertised.
Further changes should be separate PRs or issues. I'm open to any changes but we have to make sure it scales across all our projects

**Comment by leMaik** [comment]
@saschb2b 22 seconds too late :stuck_out_tongue:

**Comment by bernalrs** [comment]
Agreed, I did try to do a lot in one PR (more than it was advertised). I will say that I deemed necessary to add them to validate my code as I would have hated to introduced defects/mistakes After all, this is really why we use such tool. I will create separate pull request so you can be more familiar with the changes relating to the new tooling.

**Comment by leMaik** [comment]
> I will say that I deemed necessary to add them to validate my code as I would have hated to introduced defects/mistakes

How are linting tools supposed to find any defects introduced by adding the new props? Some unit or snapshot tests could do that instead, but unfortunately we don't have them for this project, yet. :thinking:

**Comment by bernalrs** [comment]
I think the definition of linting these days is a little more than just linting have a look at the plugins. 

```
/Users/remibernal/WebstormProjects/material-ui-search-bar/src/components/SearchBar/SearchBar.js
  71:5  error  Prop type `array` is forbidden                                                      react/forbid-prop-types
  73:5  error  Prop type `object` is forbidden                                                     react/forbid-prop-types
  83:5  error  Prop type `object` is forbidden                                                     react/forbid-prop-types
  83:5  error  propType "style" is not required, but has no corresponding defaultProp declaration  react/require-default-props

/Users/remibernal/WebstormProjects/material-ui-search-bar/src/styleguide/Wrapper.js
   5:1   error  'react-tap-event-plugin' should be listed in the project's dependencies, not devDependencies  import/no-extraneous-dependencies
   5:1   error  Expected empty line after import statement not followed by another import                     import/newline-after-import
   8:16  error  Component should be written as a pure function                                                react/prefer-stateless-function
  13:23  error  'children' is missing in props validation                                                     react/prop-types

✖ 8 problems (8 errors, 0 warnings)
```

**Comment by leMaik** [comment]
@bernalrs Okay okay, I agree that linting is useful. Now give us that awesome `disabled` prop! :smile:

**Comment by bernalrs** [comment]
I will !

**Comment by bernalrs** [comment]
Created new pull request, closing this one.

**Pull request closed without merging**